fetch/fetch.go

```diff
@@ -47,23 +49,35 @@ func init() {
 // should use this client.
 var client *http.Client
 
-// Fetcher provides a interface for anything that can get RSS data and provide it in a
+// FeedFetcher provides a interface for anything that can get RSS data and provide it in a
 // sequential fashion (e.g. without concurrency). The fetcher is the building block for
 // larger subscription routines that periodically use the fetcher to retrieve data.
-// Fetchers should therefore be treated as things that will only run inside of a single
+// FeedFetchers should therefore be treated as things that will only run inside of a single
 // thread, whereas Subscription objects are things that may run concurrently.
-type Fetcher interface {
+type FeedFetcher interface {
 	Fetch() (feed *gofeed.Feed, err error)
 }
 
-// New creates a new HTTP fetcher that can fetch rss feeds from the specified URL.
-func New(url string) Fetcher {
+// NewFeedFetcher creates a new HTTP fetcher that can fetch rss feeds from the specified URL.
+func NewFeedFetcher(url string) FeedFetcher {
 	return &httpFetcher{
 		url:    url,
 		parser: gofeed.NewParser(),
 	}
 }
 
+// HTMLFetcher is an interface for fetching the full HTML associated with a feed item
+type HTMLFetcher interface {
+	Fetch() (url string, err error)
```

Comment:
Minor typo - I think this is supposed to be `Fetch() (content string, err error)` RE the signature below?